rootfs/diff.go

```diff
@@ -39,7 +39,7 @@ func Diff(ctx context.Context, snapshotID string, sn snapshots.Snapshotter, d di
 		if err != nil {
 			return ocispec.Descriptor{}, err
 		}
-		defer sn.Remove(ctx, lowerKey)
+		defer sn.Remove(ctx, upperKey)
```

Comment:
I'm guessing this is a bug we currently have?  

cc @dmcgowan

Reply:
This is definitely a current bug.  This code path is rarely taken since diffs are usually from an active layer, so probably isn't causing anyone any issues yet. I could see buildkit or some other build system hitting this in the future.

Reply:
Split it out at least to another commit so we can possibly backport?

Reply:
ok i can do that!

Reply:
Actually, pull this into another PR so we can merge and backport.

Reply:
sounds good

Reply:
addressed in #1953